js/main.js

```diff
@@ -683,15 +691,15 @@ function initiate() {
 
         /* STAGE: checking if there is a decent conclusion*/
         var lastTextFragment = textFragments[textFragments.length-1].trim();
-        if ((lastTextFragment.length == 0 || lastTextFragment.match(/\\end\{(figure|enumerate|itemize|tabular)}\s*\\end\{solution}$/)) && textFragments[textFragments.length-2].search(/Ответ/i) < 0){
+        if ((lastTextFragment.length == 0 || lastTextFragment.match(/\\end\{(figure|enumerate|itemize|tabular)}\s*\\end\{solution}$/)) && (textFragments.length < 2 || textFragments[textFragments.length-2].search(/Ответ/i) < 0)){
             addWarning('NO_CONCLUSION');
         }
 
 
         /* STAGE: check if there are sentences starting with formula */
         for (var i = 0; i < mathFragments.length; ++i) {
             if (textFragments[i].trim().endsWith('.')) {
-                addTypicalWarning('SENTENCE_STARTS_WITH_FORMULA', 'math', i, badPos);
+                addTypicalWarning('SENTENCE_STARTS_WITH_FORMULA', 'math', i, 0);
```

Comment:
No. `badPos` is not even changed in this loop. It was used somewhere before, and was just left in scope because of JS scoping rules. This line probably was copied and pasted. And this constant will always work because of the nature of an error: this error is about starting the sentence with a formula, so it points to a beginning of this formula.